From 03eca9d92d407c71a59ff8a43067759769da0ae4 Mon Sep 17 00:00:00 2001
From: Kristian Høgsberg <krh@bitplanet.net>
Date: Tue, 27 Mar 2012 12:09:32 +0000
Subject: wayland: Stop using wl_buffer.damage

---
diff --git a/src/egl/drivers/dri2/platform_wayland.c b/src/egl/drivers/dri2/platform_wayland.c
index 0f439c7..d291f0f 100644
--- a/src/egl/drivers/dri2/platform_wayland.c
+++ b/src/egl/drivers/dri2/platform_wayland.c
@@ -598,8 +598,6 @@ dri2_swap_buffers(_EGLDriver *drv, _EGLDisplay *disp, _EGLSurface *draw)
 	    wayland_create_buffer(dri2_surf,
 		  dri2_surf->dri_buffers[__DRI_BUFFER_FRONT_LEFT]);
 
-      wl_buffer_damage(dri2_surf->wl_drm_buffer[WL_BUFFER_FRONT], 0, 0,
-		       dri2_surf->base.Width, dri2_surf->base.Height);
       wl_surface_attach(dri2_surf->wl_win->surface,
 	    dri2_surf->wl_drm_buffer[WL_BUFFER_FRONT],
 	    dri2_surf->dx, dri2_surf->dy);
--
cgit v0.9.0.2-2-gbebe
